sources/Outputs/src/ParEvent.cpp

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file  ParEvent.cpp
+ *
+ * @brief Dynaflow launcher PAR file writer for events in a contigency implementation file
+ *
+ */
+
+#include "ParEvent.h"
+
+#include "Constants.h"
+#include "Contingencies.h"
+
+#include <DYNCommon.h>
+#include <PARParameter.h>
+#include <PARParameterFactory.h>
+#include <PARParametersSetCollection.h>
+#include <PARParametersSetCollectionFactory.h>
+#include <PARReference.h>
+#include <PARReferenceFactory.h>
+#include <PARXmlExporter.h>
+#include <boost/filesystem.hpp>
+
+namespace dfl {
+namespace outputs {
+
+namespace helper {
+
+template<class T>
+static boost::shared_ptr<parameters::Parameter>
+buildParameter(const std::string& name, const T& value) {
+  return parameters::ParameterFactory::newParameter(name, value);
+}
+
+}  // namespace helper
+
+ParEvent::ParEvent(ParEventDefinition&& def) : def_{std::forward<ParEventDefinition>(def)} {}
+
+void
+ParEvent::write() {
+  using Type = dfl::inputs::Contingencies::ElementType;
+  parameters::XmlExporter exporter;
+
+  auto parametersSets = parameters::ParametersSetCollectionFactory::newCollection();
+  for (auto e = def_.contingency->elements.begin(); e != def_.contingency->elements.end(); ++e) {
```

Comment:
Use range loop instead

Reply:
Done. Also changed the` if-else` sequence for a `switch`, for clarity.